pgjdbc/src/main/java/org/postgresql/core/ConnectionFactory.java

```diff
@@ -49,7 +49,7 @@ public static QueryExecutor openConnection(HostSpec[] hostSpecs,
       Properties info) throws SQLException {
     String protoName = PGProperty.PROTOCOL_VERSION.getOrDefault(info);
 
-    if (protoName == null || protoName.isEmpty() || "3".equals(protoName)) {
+    if (protoName == null || protoName.isEmpty() || protoName.startsWith("3")) {
```

Comment:
Should we restrict the supported protocol versions only? If a newer protocol adds something in 3.7, the old existing drivers won't support 3.7 features anyway.

It looks it would be nice if the driver could reject unsupported values, so the users know the value they configure is ignored or unsupported.

WDYT?

Reply:
Probably, will fix